lib/algolia/index.rb

```diff
@@ -264,6 +264,72 @@ def get_objects(objectIDs, attributes_to_retrieve = nil, request_options = {})
       client.post(Protocol.objects_uri, { :requests => requests }.to_json, :read, request_options)['results']
     end
 
+    #
+    # Find object by the given condition.
+    #
+    # Options can be passed in request_options body:
+    #  - query (string): pass a query
+    #  - paginate (bool): choose if you want to iterate through all the
+    # documents (true) or only the first page (false). Default is true.
+    # The function takes a block to filter the results from search query
+    # Usage example:
+    #  index.find_object({'query' => '', 'paginate' => true}) {|obj| obj.key?('company') and obj['company'] == 'Apple'}
+    #
+    # @param request_options contains extra parameters to send with your query
+    #
+    def find_object(request_options = {})
+      query = ''
+      paginate = true
+      page = 0
+
+      query = request_options[:query] || request_options['query']
+      request_options.delete(:query)
+      request_options.delete('query')
+
+      paginate = request_options[:paginate] || request_options['paginate']
+      request_options.delete(:paginate)
+      request_options.delete('paginate')
+
+      while true
+        request_options['page'] = page
+        res = search(query, request_options)
+
+        res['hits'].each_with_index do |hit, i|
+          if yield(hit)
+            return {
+                'object' => hit,
+                'position' => i,
+                'page' => page,
+            }
+          end
+        end
+
+        has_next_page = page + 1 < res['nbPages']
+        if not paginate or not has_next_page
+          raise AlgoliaObjectNotFoundError.new('Object not found')
+        end
+
+        page += 1
+      end
+    end
+
+    #
+    # Retrieve the given object position in a set of results.
+    #
+    # @param [Array] res the result set to browse
+    # @param [String] object_id the object to look for
+    #
+    # @return [Integer] position of the object, or -1 if it's not in the array
+    #
+    def self.get_object_position(res, object_id)
+      res['hits'].each_with_index do |hit, i|
+        if hit['objectID'] == object_id
+          return i
+        end
+      end
+      -1
+    end
```

Comment:
It works!

I would rename `res` to something more readable like `hits`, `objects` or `records`.

Also, a more Rubyiesque way could be:
```ruby
def self.get_object_position(res, object_id)
  res['hits'].find_index { |hit| hit['objectID'] == object_id } || -1
end
```
Double check the Ruby version though :)